x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/get_prebuilt_rule_base_version/get_prebuilt_rule_base_version_handler.ts

```diff
@@ -0,0 +1,116 @@
+
+export const getPrebuiltRuleBaseVersionHandler = async (
+  context: SecuritySolutionRequestHandlerContext,
+  request: KibanaRequest<undefined, undefined, GetPrebuiltRuleBaseVersionRequest>,
+  response: KibanaResponseFactory
+) => {
+  const siemResponse = buildSiemResponse(response);
+  const { id } = request.body;
+
+  try {
+    const ctx = await context.resolve(['core', 'alerting']);
+    const soClient = ctx.core.savedObjects.client;
+    const rulesClient = await ctx.alerting.getRulesClient();
+    const ruleAssetsClient = createPrebuiltRuleAssetsClient(soClient);
+
+    const currentRule = await getRuleById({ rulesClient, id });
+
+    if (!currentRule) {
+      throw new Error(`Cannot find rule with id: ${id}`);
+    }
+
+    const [baseRule] = await ruleAssetsClient.fetchAssetsByVersion([currentRule]);
+
+    if (!baseRule) {
+      return siemResponse.error({
+        body: 'Cannot find rule base_version',
+        statusCode: 404,
+      });
+    }
+
+    const ruleDiff = calculateRuleDiff({
+      current: currentRule,
+      base: undefined, // Base version is undefined, we're using the actual base version as the target version as we want to revert the rule
+      target: baseRule,
+    });
+
+    const { diff, baseVersion, currentVersion } = formatDiffResponse(ruleDiff);
+
+    const body: GetPrebuiltRuleBaseVersionResponseBody = {
+      diff,
+      current_version: currentVersion,
+      base_version: baseVersion,
+    };
+
+    return response.ok({ body });
+  } catch (err) {
+    const error = transformError(err);
+    return siemResponse.error({
+      body: error.message,
+      statusCode: error.statusCode,
+    });
+  }
+};
+
+const formatDiffResponse = (
+  diffResult: CalculateRuleDiffResult
+): { diff: PartialRuleDiff; baseVersion: RuleResponse; currentVersion: RuleResponse } => {
+  const { ruleDiff, ruleVersions } = diffResult;
+  const installedCurrentVersion = ruleVersions.input.current;
+  const baseVersion = ruleVersions.input.target;
+  invariant(installedCurrentVersion != null, 'installedCurrentVersion not found');
+  invariant(baseVersion != null, 'baseVersion not found');
+
+  const baseRule: RuleResponse = {
+    ...convertPrebuiltRuleAssetToRuleResponse(baseVersion),
+    id: installedCurrentVersion.id,
+    // Set all fields to the original Elastic version values
+    created_at: installedCurrentVersion.created_at,
+    created_by: installedCurrentVersion.created_by,
+    updated_at: installedCurrentVersion.created_at,
+    updated_by: installedCurrentVersion.created_by,
+    revision: 0,
+  };
+
+  return {
+    baseVersion: baseRule,
+    currentVersion: installedCurrentVersion,
+    diff: {
+      fields: pickBy<ThreeWayDiff<unknown>>(
+        ruleDiff.fields,
+        (fieldDiff) =>
+          fieldDiff.diff_outcome !== ThreeWayDiffOutcome.StockValueNoUpdate &&
+          fieldDiff.diff_outcome !== ThreeWayDiffOutcome.MissingBaseNoUpdate
```

Comment:
Passing base rule version is `base` and `target` would make it sufficient to check only for `ThreeWayDiffOutcome.CustomizedValueNoUpdate` to retrieve customized fields.

Reply:
Changed over